lib/humanized_id/humanizer.rb

```diff
@@ -1,43 +1,37 @@
 # frozen_string_literal: true
 module HumanizedId
   class Humanizer
-    SOURCE_CHARSET = %w(0123456789abcdefghijklmnopqrstuvwxyz).freeze
+    SOURCE_CHARSET = (('0'..'9').to_a + ('a'..'z').to_a).freeze
 
     def initialize(id:, min_length: nil, prefix: '')
       @id               = id
       @min_length       = min_length
-      @prefix           = prefix.nil? ? '' : prefix
-      @target_charset   = HumanizedId::CHARACTERSET.join('')
-      @source_charset   = SOURCE_CHARSET.join('')
+      @prefix           = prefix.to_s
+      @target_charset   = CHARSET.join
+      @source_charset   = SOURCE_CHARSET.join
     end
 
     def generate_humanized_id
-      new_id = convert_to_target_base id: @id
-      new_id = convert_to_target_charset id: new_id
-      new_id = resize id: new_id if @min_length
+      new_id = resize convert_to_target_charset convert_to_target_base(@id)
```

Comment:
I prefer the older way - it's actually harder to read this properly (due to the reverse ordering) and we lose named parameters in the process.